### ISSUE TITLE
Continued high CPU utilization

### ISSUE THREAD

**Issue by jgclark** [ttscoff/bunch#112]
macOS 11.4 Intel on MacMini (2018)
bunch 1.4.1 beta

I noticed the fan noise and investigated. Bunch has been sat at almost exactly 100% for 5+ minutes.  It's not apparently doing anything. And the last log entry was over an hour ago (as expected).
I also notice that CPU time is 1:21:59 since last boot (10 days ago).

Quitting bunch and restarting brings it back to 0%.

Just realised it would have been more helpful to sample the process before quitting the app, sorry.

**Comment by ttscoff** [comment]
So this is hard to diagnose. I run bunch all day every day on an M1 Mac mini and CPU usage is usually at 0% (unless a Bunch is opening and spawning processes). I don't have an Intel machine handy that can test with 11.4. A sample dump may help, but it may not offer an obvious culprit.

- Brett

**Comment by jgclark** [comment]
Understood. It was more for completeness, and to put something here that others can add to if they see similar.  But equally happy for you to close if you wish.

**Comment by ttscoff** [comment]
Oh, no, I wasn't implying this shouldn't be here, just stating that I don't have anything to go off of at this point. It absolutely should be here and if anyone can add more data, it will be very useful.

- Brett

**Comment by yverrette** [comment]
Just want to confirm I have the same thing happening on an Intel 2016 MacBook Pro. Not much to offer in terms of new information but happy to run any diagnostic you might need.

**Comment by ttscoff** [comment]
The best troubleshooting I can come up with for this is just to open Console.app to the system log and turn on streaming while it's happening. I probably don't need to see the whole thing, but you can filter for "Bunch" and see if there are any related messages happening.

If you open Activity Monitor and locate Bunch in the list of apps, you can also use the ... button at the top to generate a sample/spin dump. Might be useful if you want to attach. If you're going to post large logs, please [create a Gist](https://gist.github.com) and just put the link in here.

**Comment by yverrette** [comment]
Given the content of the spindump and console log, I preferrer not to share on the public forum but I sent you the info to get them.

**Comment by ttscoff** [comment]
@yverrette Got your spindump. What exactly was happening at the time this was generated? Was Bunch idle at the time? The symbols indicate that it was running (or attempting to run) a shell script. Is it possible a Bunch got stuck executing a script that never completed?

Also, what's your default shell? Bash/Zsh, or something else?

**Comment by ttscoff** [comment]
@yverrette Also noting references to mdfind in your console log. Unless I'm forgetting something, Bunch only accesses mdfind if explicitly told to do so using tag or search syntax in a file line. Are you using either method at all?

There's an error that mdfind requires Address Book and Calendar permissions, and I may need to actually specify that I need those in order to allow users to grant permission. If that's the case, I'm just surprised this hasn't popped up before. Not sure it would cause the background CPU usage specific to this ticket, but could definitely cause some errors.

**Comment by yverrette** [comment]
If I recall correctly I did an OS restart and the spindump was started right before I triggered a bunch. That bunch is sequential and does a few things: start a few apps, open a few web pages and pass some keystrokes to websites in Safari. It does not run a shell script file but it does pass a shell command to iTerm and you are right, it waits for that command to complete. 

That said, I did some more digging and found my specific issue. I use [Unite 4](https://www.bzgapps.com/unite) to have the Amazon AWS web console behave like a Mac app. For some reason in my original script, the only way I got it to run and communicate using Bunch was to escape spaces the application name. (i.e. `AWS - some company` became `AWS\ -\ some\ company` ). This worked to launch the app but it seems like it did not properly communicate commands (sleep time and keystrokes) to it. As these commands never came back, it looks like it was stuck in a loop waiting for some return. I removed the escaping and it seems to work ok now on 1.4.4 (118). I don't know if something changed in Bunch since I originally wrote the script mid July.  I also updated Unite 4 last week which could have had an impact as I remember this AWS web console application not launching from Alfred until I updated Unite 4.

As for the mdfind I have no clue why it would be in play. As I use Alfred as a luncher, not sure why spotlight would be running. I guess it's possible Alfred uses the spotlight indexes and I could have triggered it while the bunch was running.
